internal/server/server.go

```diff
@@ -169,13 +171,13 @@ func New(options Options) (*Server, error) {
 		return nil, fmt.Errorf("postgres dsn: %w", err)
 	}
 	options.DB.DSN = dsn
+	options.DB.RootKeyFilePath = options.DBEncryptionKey
 
-	dbKeyProvider, ok := server.keys[options.DBEncryptionKeyProvider]
-	if !ok {
-		return nil, fmt.Errorf("key provider %s not configured", options.DBEncryptionKeyProvider)
+	if _, err := os.Stat(options.DB.RootKeyFilePath); errors.Is(err, fs.ErrNotExist) {
+		if err := encrypt.CreateRootKey(options.DB.RootKeyFilePath); err != nil {
+			return nil, err
+		}
```

Comment:
This initializes a new root key when it's missing. This used to work at some point, but when I wrote the end-to-end test suite it wasn't working, so I had to do it explicitly here:  https://github.com/infrahq/infra/blob/c5b9fa43d6d910b5d55a75cd5a74c5ffeb6a7faf/.github/workflows/ci-core.yaml#L139

I'm not sure what was not working. If we want to initialize a key, it should be part of the server process startup logic. So I've added this here to restore the initializing of a root key when it doesn't exist.